### PR TITLE
Support for IP families in the unified image

### LINKED ISSUE #1031: Specifying IP families when using the unified image
The feature to specify an IP family to select from the PodIPs list does not work with the unified image. We should add support for this mechanism.

### PR DESCRIPTION
**Pull request by brownleej**
# Description

*Please include a summary of the change and which issue is addressed. If this change resolves an issue, please include the issue number in the description.*

This adds support for the IP family argument type when using the unified image. This required pulling in new models for the monitor conf, which in turn required updating a bunch of dependencies due to a major update to go-logr/logr.

This PR is currently built on an in-development version of the fdbkubernetesmonitor dependency. Before merging, we need to update this to use the build from main.

Fixes #1031 

## Type of change

*Please select one of the options below.*

- New feature (non-breaking change which adds functionality)

## Discussion

*Are there any design details that you would like to discuss further?*

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

## Documentation

*Did you update relevant documentation within this repository?*

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

## Follow-up

*Are there any follow-up issues that we should pursue in the future?*

*Does this introduce new defaults that we should re-evaluate in the future?*

### CODE REVIEW

File: internal/monitor_conf.go [hunk 1]
```diff
@@ -267,7 +268,7 @@ func GetMonitorProcessConfiguration(cluster *fdbv1beta2.FoundationDBCluster, pro
 }
 
 // buildIPArgument builds an argument that takes an IP address from an environment variable
-func buildIPArgument(parameter string, environmentVariable string, imageType FDBImageType, sampleAddresses []fdbv1beta2.ProcessAddress) []monitorapi.Argument {
+func buildIPArgument(parameter string, environmentVariable string, imageType FDBImageType, sampleAddresses []fdbv1beta2.ProcessAddress, podIpFamily *int) []monitorapi.Argument {
```
Comment (johscheuer) [inline internal/monitor_conf.go:271]:
```suggestion
func buildIPArgument(parameter string, environmentVariable string, imageType FDBImageType, sampleAddresses []fdbv1beta2.ProcessAddress, podIPFamily *int) []monitorapi.Argument {
```

File: internal/pod_client.go [hunk 2]
```diff
@@ -323,7 +323,7 @@ func (client *realFdbPodSidecarClient) updateDynamicFiles(filename string, conte
 func (client *realFdbPodAnnotationClient) GetVariableSubstitutions() (map[string]string, error) {
 	environmentData, present := client.Pod.Annotations[EnvironmentAnnotation]
 	if !present {
-		client.logger.Info("Waiting for Kubernetes monitor to update annotations", "annotation", EnvironmentAnnotation)
+		client.logger.Info("Waiting for Kubernetes monitor to update annotations", "annotation", EnvironmentAnnotation, "jpb-annotations", client.Pod.Annotations)
```
Comment (johscheuer) [inline internal/pod_client.go:326]: I guess the latter was for debugging?

**Review by johscheuer — COMMENTED** [review]
Changes LGTM 👍

File: internal/pod_client.go (hunk 2 above)
Comment (brownleej) [reply internal/pod_client.go]: Yup, removing that now.

**Review by johscheuer — COMMENTED** [review]
Given that we had some moving pieces since the PR was opened, do you think it make sense to close this one and reopen it once we have more time to actually implement this feature?

**Pull request closed without merging**